File: models/java/BuyerAgent.java
/*
 * BuyerAgent class
 *
 * CSS 610	
 * George Mason University			
 * Spring 2005
 *
 * Rob Axtell
 *
 */

import java.util.concurrent.*;

public class BuyerAgent {

    private double value;
    private boolean traded;
    public double price;
    public Semaphore lock;

    public BuyerAgent(double maxBuyerValue) {
        value = ThreadLocalRandom.current().nextDouble(0, 1) * maxBuyerValue;
        traded = false;
	lock = new Semaphore(1);
    }

    public void setTraded(boolean t) {
        traded = t;
    }

    public boolean hasTraded() {
        return (traded);
    }

    public double formBidPrice() {
        //return Math.random() * value;
        return ThreadLocalRandom.current().nextDouble(0, 1) * value;
    }
}

File: models/scala/src/main/scala/BuyerAgent.scala
package ZIMarketParallel

import java.util.concurrent._

class BuyerAgent(maxBuyerValue: Double) {


  //private var cost: Double = ThreadLocalRandom.current().nextDouble(0, 1) * maxCost
  private var value: Double = ThreadLocalRandom.current().nextDouble(0, 1) * maxBuyerValue

  private var traded: Boolean = false

  var price: Double = _

  var lock: Semaphore = new Semaphore(1)

  def setTraded(t: Boolean) {
    traded = t
  }

  def hasTraded(): Boolean = (traded)

  //def formBidPrice(): Double = Math.random() * value
  def formBidPrice(): Double = ThreadLocalRandom.current().nextDouble(0, 1) * value
}